packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart

```diff
@@ -464,11 +465,44 @@ Information about project "Runner":
     expect(info.schemeFor(const BuildInfo(BuildMode.debug, 'unknown', treeShakeIcons: false)), isNull);
   });
 
+  testWithoutContext('reports default scheme error and exit', () {
+    final XcodeProjectInfo defaultInfo = XcodeProjectInfo(
+      <String>[],
+      <String>[],
+      <String>['Runner'],
+      logger,
+    );
+
+    expect(
+      () => defaultInfo.reportFlavorNotFoundAndExit(),
```

Comment:
technically, you could pass in `defaultInfo.reportFlavorNotFoundAndExit` as a tear-off, since the throws matchers just require a zero-arity function. Just an FYI

Reply:
Audited, got a PR coming your way.